R/function_argument_linter.R

```diff
@@ -1,6 +1,30 @@
 #' Function argument linter
 #'
-#' Check that arguments with defaults come last in all function declarations, as per the tidyverse design guide.
+#' Check that arguments with defaults come last in all function declarations,
+#' as per the tidyverse design guide.
+#'
```

Comment:
Might be worth noting how we approached a fix to be maximally backwards-compatible for existing code: set these arguments to be `NULL` by default. See also #1546.

Maybe (1) mention this in exposition here and (2) suggest a `x=NULL` version (etc) under the `okay` examples

WDYT?

Reply:
Excellent points. I have both updated the description and added an example for this.